src/components/Table/index.tsx

```diff
@@ -78,150 +79,82 @@ interface TableProps {
-        };
-    }
-
-    public componentDidMount() {
-        const { filters } = this.props;
-        if (filters && filters.length > 0) {
-            this.handleFilter(filters[0]);
-        }
-    }
-
-    public componentWillReceiveProps(next: TableProps) {
-        if (this.state.selectedRowKey !== next.selectedKey) {
-            this.setState({selectedRowKey: next.selectedKey});
-        }
-    }
-
-    public componentDidUpdate(prevProps: TableProps) {
-        if (prevProps.data !== this.props.data && this.props.filters) {
-            const activeFilter = this.props.filters.find(
-                filter => filter.name === this.state.activeFilter,
-            );
-
-            if (activeFilter) {
-                this.handleFilter(activeFilter);
-            }
-        }
-    }
-
-    public render() {
-        const { data, header, titleComponent, filters = [], rowKeyIndex } = this.props;
-
-        this.ensureDataIsValid(data);
-
-        const cn = classNames('cr-table-header__content', {
-            'cr-table-header__content-empty': !titleComponent && filters.length === 0,
-        });
-
-        return (
-            <div className="cr-table-container">
-                <div className={cn}>
-                    {titleComponent ? this.renderTitleComponent() : null}
-                    {filters.length
-                        ?
-                        <div className="cr-table__filters">{this.renderFilters()}</div>
-                        : null}
-                </div>
-                <table className={'cr-table'}>
-                    {header && header.length && this.renderHead(header)}
-                    {this.renderBody(data, rowKeyIndex)}
-                </table>
-                {this.renderBackground(data)}
-            </div>
-        );
-    }
-
-    private renderTitleComponent() {
-        const { titleComponent } = this.props;
-
-        return <div className={'cr-title-component'}>{titleComponent}</div>;
-    }
-
-    private renderRowCells(row: CellData[]) {
-
+export const Table: React.FC<TableProps> = (props: TableProps) => {
+    const { formatMessage } = useIntl();
+    const [activeFilter, setActiveFilter] = React.useState<TableState['activeFilter']>(undefined);
+    const [resultData, setResultData] = React.useState<TableState['resultData']>(undefined);
+    const [selectedRowKey, setSelectedRowKey] = React.useState<TableState['selectedRowKey']>(props.selectedKey);
+
+    const {
+        data,
+        header,
+        titleComponent,
+        filters = [],
+        rowKeyIndex,
+        onSelect,
+        rowBackground,
+        side,
+        rowBackgroundColor = 'rgba(184, 233, 245, 0.7)',
+    } = props;
+
+    const cn = React.useMemo(() => classNames('cr-table-header__content', {
+        'cr-table-header__content-empty': !titleComponent && filters.length === 0,
+    }), [titleComponent, filters.length]);
+
+    const renderRowCells = React.useCallback((row: CellData[]) => {
```

Comment:
Use react.memo instead

Reply:
I'm not going to change logic which was written previously. This PR is about hooks

Reply:
this is the function with parameters which can be changed

Reply:
yes and here we are rendering element. So in this case I think better to compare the result of the function instead of the params

Reply:
This logic I have not touched because its written before and it's works somehow. There are many more important structural tasks. So, lets not play around these small cases and put this PR ahead. This part of code may disappear after lower lever of refactoring.